File: src/ros2_bridge/mappers/foxglove_msgs/vector3.rs
//! Typed mapper for `foxglove_msgs/msg/Vector3`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn vector3_to_bus(msg: ros_env::foxglove_msgs::msg::Vector3) -> crate::foxglove_msgs::msg::v1::Vector3 {
    crate::foxglove_msgs::msg::v1::Vector3 {
        x: msg.x,
        y: msg.y,
        z: msg.z,
    }
}

pub(crate) fn vector3_to_ros(bus: crate::foxglove_msgs::msg::v1::Vector3) -> ros_env::foxglove_msgs::msg::Vector3 {
    ros_env::foxglove_msgs::msg::Vector3 {
        x: bus.x,
        y: bus.y,
        z: bus.z,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct FoxgloveMsgsVector3Mapper;

impl TypedTopicMapper for FoxgloveMsgsVector3Mapper {
    type Ros = ros_env::foxglove_msgs::msg::Vector3;
    type Bus = crate::foxglove_msgs::msg::v1::Vector3;

    fn type_name(&self) -> &'static str {
        "foxglove_msgs/msg/Vector3"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(vector3_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(vector3_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/foxglove_msgs/vector3.py
"""Generated mapper for `foxglove_msgs/msg/Vector3`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.foxglove_msgs.msg.v1 import Vector3 as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def vector3_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.x = msg.x
    bus.y = msg.y
    bus.z = msg.z
    return bus


def vector3_to_ros(bus):
    from foxglove_msgs.msg import Vector3 as RosMsg

    out = RosMsg()
    out.x = bus.x
    out.y = bus.y
    out.z = bus.z
    return out


class FoxgloveMsgsVector3Mapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from foxglove_msgs.msg import Vector3 as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return vector3_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return vector3_to_ros(bus)
